model_analyzer/result/run_config_measurement.py

```diff
@@ -0,0 +1,373 @@
+
+        self._model_config_measurements = []
+        self._model_config_weights = []
+
+    @classmethod
+    def from_dict(cls, run_config_measurement_dict):
+        run_config_measurement = RunConfigMeasurement(None, {})
+
+        run_config_measurement._key = run_config_measurement_dict['_key']
+        run_config_measurement._gpu_data = run_config_measurement_dict[
+            '_gpu_data']
+        run_config_measurement._avg_gpu_data = run_config_measurement_dict[
+            '_avg_gpu_data']
+        run_config_measurement._model_config_measurements = run_config_measurement_dict[
+            '_model_config_measurements']
+
+        return run_config_measurement
+
+    def set_model_config_weighting(self, model_config_weights):
+        """
+        Sets the model config weightings used when calculating 
+        weighted metrics
+        
+        Parameters
+        ----------
+        model_weights: list of ints
+        Weights are the relative importance of the model_configs 
+        with respect to one another
+        """
+        self._model_config_weights = [
+            model_config_weight / sum(model_config_weights)
+            for model_config_weight in model_config_weights
+        ]
+
+    def add_model_config_measurement(self, model_config_name,
+                                     model_specific_pa_params, non_gpu_data):
+        """ 
+        Adds a measurement from a single model config in this PA's run 
+        
+        model_config_name : string
+            The model config name that was used for this PA run
+        model_specific_pa_params: dict
+            Dictionary of PA parameters that change between models
+            in a multi-model run
+        non_gpu_data : list of Records
+            Metrics that do not have a GPU UUID associated with them,
+            from either CPU or PA
+        """
+        self._model_config_measurements.append(
+            ModelConfigMeasurement(model_config_name, model_specific_pa_params,
+                                   non_gpu_data))
+
+    def set_metric_weightings(self, metric_objectives):
+        """
+        Sets the metric weighting for all non-GPU measurements
+        
+        Parameters
+        ----------
+        metric_objectives : list of dict of RecordTypes
+            One entry per ModelConfig
+        """
+        for index, measurement in enumerate(self._model_config_measurements):
+            measurement.set_metric_weighting(metric_objectives[index])
+
+    def key(self):
+        """
+        Returns
+        -------
+        Key used to uniquely identify the RunConfigMeasurement
+        """
+
+        return self._key
+
+    def data(self):
+        """
+        Returns
+        -------
+        per model list of a list of Records
+            All the metric values in each model's measurement 
+            for both GPU and non-GPU
+        """
+
+        return [
+            self._avg_gpu_data + mcm.non_gpu_data()
```

Comment:
remind me what the avg_gpu_data is for?

Reply:
If running on multiple GPUs we average the GPU stats across the them and report this to the user.